### PR TITLE
[WIP] Replace react-collapse with accessible, homegrown alternative

### PR DESCRIPTION
**Pull request by dananjohnson**

### CODE REVIEW

**Comment by dananjohnson** [comment]
@1aurend if you're curious, you can see how I handled the notification form in 18457c4. There's some background on the approach in the commit message. Let me know if you see any issues!

I also squashed all the commits down into a cleaner commit history.

File: client/src/backend/components/project/hero/Builder/BlockHeaderDetail.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  titleId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  description: PropTypes.string,
+  hasDescription: PropTypes.bool
+};
+
+BlockHeaderDetail.propTypes = {
+  titleId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  title: PropTypes.string.isRequired,
+  description: PropTypes.string
+};
```
Comment (1aurend) [inline client/src/backend/components/project/hero/Builder/BlockHeaderDetail.js:35]: @dananjohnson Before you merge this I noticed, the duplicate `propTypes` here. Totally something I did. 🙃 The second set, sans `hasDescription` is correct.

**Comment by 1aurend** [comment]
Nice @dananjohnson. The notifications form looks great! Thanks for sending this my way.

File: client/src/backend/components/project/hero/Builder/BlockHeaderDetail.js (hunk 1 above)
Comment (dananjohnson) [reply client/src/backend/components/project/hero/Builder/BlockHeaderDetail.js:35]: Good eye! Will fix then merge.

**Pull request merged**